fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java

```diff
@@ -1335,7 +1336,7 @@ private void addPartitions(Database db, String tableName, List<PartitionDesc> pa
             // get distributionInfo
             distributionInfo = getDistributionInfo(olapTable, addPartitionClause);
 
-            if (distributionInfo.getBucketNum() == 0) {
+            if (distributionInfo.getBucketNum() == 0 || config.enable_auto_create_tablet) {
```

Comment:
No matter what `enable_auto_create_tablet ` is , calAvgBucketNumOfRecentPartitions will be called.